qiskit_experiments/library/tomography/tomography_analysis.py

```diff
@@ -153,254 +152,73 @@ def _run_analysis(self, experiment_data):
-        fitter_options: dict = None,
-    ):
-        """Run tomography fitter on tomography data"""
-        # Construct default values for qubit options if not provided
-        if preparation_qubits is None:
-            preparation_qubits = tuple(range(preparation_data.shape[1]))
-        if measurement_qubits is None:
-            measurement_qubits = tuple(range(measurement_data.shape[1]))
-
-        # Get dimension of the preparation and measurement qubits subsystems
-        prep_dims = (1,)
-        if preparation_qubits:
-            if not preparation_basis:
-                raise AnalysisError("No tomography preparation basis provided.")
-            prep_dims = preparation_basis.matrix_shape(preparation_qubits)
-        meas_dims = (1,)
-        full_meas_qubits = measurement_qubits
-        if measurement_qubits:
-            if conditional_measurement_indices is not None:
-                # Remove conditional qubits from full meas qubits
-                full_meas_qubits = [
-                    q
-                    for i, q in enumerate(measurement_qubits)
-                    if i not in conditional_measurement_indices
-                ]
-            if full_meas_qubits:
-                if not measurement_basis:
-                    raise AnalysisError("No tomography measurement basis provided.")
-                meas_dims = measurement_basis.matrix_shape(full_meas_qubits)
-
-        if full_meas_qubits:
-            # QPT or QST
-            input_dims = prep_dims
-            output_dims = meas_dims
-        else:
-            # QST of POVM effects
-            input_dims = meas_dims
-            output_dims = prep_dims
-
-        # Use preparation dim to set the expected trace of the fitted state.
-        # For QPT this is the input dimension, for QST this will always be 1.
-        trace = np.prod(prep_dims) if self.options.rescale_trace else None
-
-        # Get tomography fitter options
-        if fitter_options is None:
-            fitter_options = {}
-
-        # Work around to set proper trace and trace preserving constraints for
-        # cvxpy fitter
-        if fitter in self._cvxpy_fitters:
-            fitter_options = fitter_options.copy()
-
-            # Add default value for CVXPY trace constraint if no user value is provided
-            # Use preparation dim to set the expected trace of the fitted state.
-            # For QPT this is the input dimension, for QST this will always be 1.
-            if "trace" not in fitter_options:
-                fitter_options["trace"] = trace
-
-            # By default add trace preserving constraint to cvxpy QPT fit
-            if preparation_data.shape[1] > 0 and "trace_preserving" not in fitter_options:
-                fitter_options["trace_preserving"] = True
-
-        # Run tomography fitter
-        t_fitter_start = time.time()
         try:
-            fit, fitter_metadata = fitter(
+            fits, fitter_metadata = fitter(
```

Comment:
I like new implementation since now it's clear that the fit is not intended to be overruled by the user subclass.